servers/visual/visual_server_scene.cpp

```diff
@@ -1936,6 +1946,33 @@ void VisualServerScene::_prepare_scene(const Transform p_cam_transform, const Ca
 
 			ins->depth = near_plane.distance_to(ins->transform.origin);
 			ins->depth_layer = CLAMP(int(ins->depth * 16 / z_far), 0, 15);
+
+			// if lod is active, and the instance is not within its lod range, don't render it
+			if (ins->lod_begin > 0.f || ins->lod_end > 0.f) { // lod valid
+				float lod_begin_with_hys = ins->lod_begin;
+				float lod_end_with_hys = ins->lod_end;
+				if (ins->prev_lod_state) {
+					lod_begin_with_hys -= ins->lod_begin_hysteresis / 2.f;
+					lod_end_with_hys   += ins->lod_end_hysteresis / 2.f;
+				} else {
+					lod_begin_with_hys += ins->lod_begin_hysteresis / 2.f;
+					lod_end_with_hys   -= ins->lod_end_hysteresis / 2.f;
+				}
+
+				if (ins->lod_begin <= 0.f) {
+					lod_begin_with_hys = -std::numeric_limits<float>::infinity();
```

Comment:
We don't use `std` in Godot, you should use `Math_INF` from `core/math/math_funcs.h`.